user/overriding-packages.rst

```diff
@@ -0,0 +1,282 @@
+It should also ignore entry points that can't be loaded.
+
+Here's how to do it:
+
+.. code-block:: Python
+
+    from importlib.metadata import entry_points
+
+    # Deduplicate entry point specifications before loading
+    deduplicated_entry_points = {}
+    # When faced with duplicates, this loop keeps the last entry point found
+    for ep in entry_points()['your_group_name']:
+        deduplicated_entry_points[ep.name] = ep
+
+    for ep in deduplicated_entry_points:
+        try:
+            inst = ep.load()
+        except ImportError:
+            # Ignore entry point specifications that can't be loaded
+            pass
+
+All Known issues
+----------------
+
+Include Directory Search Order Problem
+**************************************
+
+When overriding a package, it's possible for a package to find its headers from the underlay instead of the overlay.
+This may cause a failure to build or undefined behavior at runtime depending on the differences between those headers.
+
+Consider an overlay containing package ``foo`` and ``bar``, and an underlay containing ``bar`` and ``baz``.
+``foo`` depends on ``bar`` and ``baz``.
+Say the underlay is a merged workspace, and both the overriden ``bar`` and ``baz`` install their headers to a directory called ``include/``.
+If any libraries or executables in ``foo`` are configured to search for headers in ``baz``'s include directory first, then headers from overridden ``bar`` will also be found first.
+
+When it can happen
+++++++++++++++++++
+
+* The underlay workspace is a merged workspace
+* The overridden package installs header files (C/C++)
+* The overriding package's headers are different from the overridden package's
+* Another package in the underlay is not overridden and installs headers to the same directory as the overridden package (such as ``include``)
+* A package in the overlay depends on both the package being overridden and the mentioned additional package in the underlay
+
+How to avoid it
++++++++++++++++
+
+Use isolated workspaces
+^^^^^^^^^^^^^^^^^^^^^^^
+
+If your underlay is an isolated workspace, then no two packages in it will have the same include directory.
+Using an isolated overlay workspace won't help if your underlay is already a merged workspace (for example, the default ROS installation when installed from binary packages).
+
+Sort include directories according to the workspace order
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+One solution is to sort include directories before passing them to the compiler such that headers are found in overlay workspaces before underlays.
+This requires making the build system for every package in an overlay aware of workspaces and the order they were sourced.
+The only known implementation of this is in ``catkin`` in ROS 1.
```

Comment:
We should elaborate here. I'd be willing to bet that some will walk away thinking "If catkin did it, why doesn't colcon?".

Reply:
I added a little more context to hopefully distance it from `catkin` the build tool in 7055c8a